### DISCUSSION TITLE
fit and finish

### DISCUSSION THREAD

**Discussion by matthew-n** [tcartwright/SqlServer.Rules#14] [General]
Looks like your doing XML doc in the CS files. 

Do you want to ticket out the work ?
Do we want to build this in a GitHub action ( Think it has to be .Net SDK project) ?

**Comment by tcartwright** [comment]
We can ticket out the work. 

- powershell to extract the MD from the C# attributes, and xml comments. 
- Convert projects to SDK style (This works for all except the DB project, unless we use [SDK Style SqlProj](https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/issues/170))
- Add c# comments to the classes.
    - Fill out summary tag
    -  Add / Edit FriendlyName, and IsIgnorable tags
    -  Add ExampleMd tag with good / bad examples

**Reply by tcartwright** [reply]
Oh, and lastly. Drop the csv. As we will use the source to generate the MD.

**Reply by matthew-n** [reply]
Ok, looking to chip in. I'll make the tickets, if you haven't, after lunch.

**Reply by tcartwright** [reply]
We can parcel out the xml comments. You and Walter can figure out the github action. I'm gonna try working the format action.

**Comment by tcartwright** [comment]
Is there a way to apply a formatter to commits? I have a real bad habit of formatting entire files when I work on them. Which has bloated some of my commits. 

https://gist.github.com/EtherZa/581d9276336353838b2c939f9554d479  ???

**Reply by matthew-n** [reply]
Probably want to add it to the VS Proj, as I don't think you can force precommit hooks. Then one big format the world commit. And will be able to see ppl changing away from the standard and refuse the commits.

**Comment by matthew-n** [comment]
Also an installer even if it's your PS script added to the root folder.

**Reply by tcartwright** [reply]
I'll work the installer. It will be a powershell script.

**Comment by tcartwright** [comment]
https://www.meziantou.net/enforce-dotnet-code-style-in-ci-with-dotnet-format.htm

**Comment by tcartwright** [comment]
I was able to convert the projects to SDK style using try-convert.

https://docs.microsoft.com/en-us/dotnet/architecture/modernize-desktop/example-migration#:~:text=Migrate%20with%20Try%20Convert%20tool

https://github.com/dotnet/try-convert/releases

**Reply by matthew-n** [reply]
would you open a branch and add just that commit with something like `part of #17` in the message

**Reply by tcartwright** [reply]
Ooops, too late, already committed  :-P

**Comment by delgertome** [comment]
Here's the initial results of the document generator: https://github.com/tcartwright/SqlServer.Rules/blob/master/docs/TableOfContents.md

Let me know what y'all think.

**Reply by delgertome** [reply]
My first thoughts is to change TableOfContents.md to README.MD so that it shows by default when opening the docs directory. Any thoughts/opposition to that?
